lib/config/convict.js

```diff
@@ -0,0 +1,835 @@
+    },
+    disableRequestedAuthnContext: {
+      doc: 'SAML if truthy, do not request a specific authentication context. This is known to help when authenticating against Active Directory (AD FS) servers.',
+      format: 'Boolean',
+      env: 'CMD_SAML_DISABLEREQUESTEDAUTHNCONTEXT',
+      default: false
+    },
+    groupAttribute: {
+      format: '*',
+      env: 'CMD_SAML_GROUPATTRIBUTE',
+      default: undefined
+    },
+    externalGroups: {default: []},
+    requiredGroups: {default: []},
+    attribute: {
+      id: {
+        format: '*',
+        env: 'CMD_SAML_ATTRIBUTE_ID',
+        default: undefined
+      },
+      username: {
+        format: '*',
+        env: 'CMD_SAML_ATTRIBUTE_USERNAME',
+        default: undefined
+      },
+      email: {
+        format: '*',
+        env: 'CMD_SAML_ATTRIBUTE_EMAIL',
+        default: undefined
+      }
+    }
+  },
+  email: {
+    doc: 'Allow users to sign-in using their E-Mail address.',
+    default: true,
+    env: 'CMD_EMAIL',
+    format: 'Boolean'
+  },
+  allowEmailRegister: {
+    doc: 'Allow users to sign up using their E-Mail. Requires email to be set to true.',
+    default: true,
+    env: 'CMD_ALLOW_EMAIL_REGISTER',
+    format: 'Boolean'
+  },
+  allowGravatar: {
+    doc: 'Allow the usage of Libravatar for user profile pictures.',
+    default: true,
+    env: 'CMD_ALLOW_GRAVATAR',
+    format: 'Boolean'
+  },
+  allowPDFExport: {
+    doc: 'Allow PDF export documents as PDF.',
+    default: true,
+    env: 'CMD_ALLOW_PDF_EXPORT',
+    format: 'Boolean'
+  },
+  openID: {
+    doc: 'Allow OpenID as authentication method.',
+    default: false,
+    env: 'CMD_OPENID',
+    format: 'Boolean'
+  },
+  linkifyHeaderStyle: {
+    doc: 'Default linking style for headers used by the markdown rendering engine for notes',
+    format: function (val) {
+      if (!/(keep-case|gfm)/.test(val)) {
+        throw new Error('Value has to be \'keep-case\' or \'gfm\'')
+      }
+    },
+    default: 'keep-case'
+  },
+  version: {
+    doc: 'Version number of CodiMD',
+    format: 'semver',
+    default: null
+  },
+  minimumCompatibleVersion: {
+    doc: 'Versio number of compatible CodiMD frontends',
+    format: 'semver',
+    default: '1.1.0'
+  },
+  fullversion: {
+    doc: 'Full version number of CodiMD (might contains a commit hash)',
+    format: 'semver',
+    default: null
+  },
+  sourceURL: {
+    doc: 'Link to the source code repository',
+    format: 'http-url',
+    default: 'https://github.com/codimd/server'
+  },
+  allowedUploadMimeTypes: {
+    doc: 'MimeTypes that are allowed to be uploaded as images',
+    format: 'Array',
+    default: [
+      'image/jpeg',
+      'image/png',
+      'image/jpg',
+      'image/gif',
+      'image/svg+xml'
+    ]
+  },
+  isFacebookEnable: {
+    default: false
+  },
+  isGoogleEnable: {
+    default: false
+  },
+  isDropboxEnable: {
+    default: false
+  },
+  isTwitterEnable: {
+    default: false
+  },
+  isEmailEnable: {
```

Comment:
could we auto-enable these (if they are not already auto-enabled) just by having any form of configuration on any of its values?

Reply:
We have the actual config variables for that above. `isEmailEnable` is actually just a internal state that is not actually configured by admins. That's done through `email: true`, which is by default enabled.

Those variables basically exists for now as part of the convict config to keep the current application logic as it is and not having the need to rewrite half of that with a separate state library.

Reply:
OK, got it :-)
Then I'd request adding a line about that as a comment before all of the `is....Enable` statements.